test/e2e/util.go

```diff
@@ -675,3 +755,36 @@ func NewClientConfigForTest(t *testing.T) *rest.Config {
 	require.NoError(t, err)
 	return config
 }
+
+func WaitForClusterOperatorStatus(t *testing.T, client configv1client.ConfigV1Interface, available, progressing, degraded *bool) error {
+	status := map[configv1.ClusterStatusConditionType]bool{} // struct for easy printing the conditions
+	return wait.PollImmediate(time.Second, 5*time.Minute, func() (bool, error) {
+		clusterOperator, err := client.ClusterOperators().Get(context.TODO(), "authentication", metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			t.Logf("clusteroperators.config.openshift.io/authentication: %v", err)
+			return false, nil
+		}
+		if retry.IsHTTPClientError(err) {
+			t.Logf("clusteroperators.config.openshift.io/authentication: %v", err)
+			return false, nil
+		}
+		availableStatusIsMatch, progressingStatusIsMatch, degradedStatusIsMatch := true, true, true
```

Comment:
The logic seems to be correct but do you think using this [helper](https://github.com/openshift/library-go/blob/master/pkg/config/clusteroperator/v1helpers/status.go#L121) would simplify it even further?

Reply:
I think the current logic is the simplest for watching a subset of all the conditions while also being able to print the current operator status, but maybe you have even a simpler solution that I'm not seeing?

Reply:
maybe something similar like this? https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/test/library/cluster_operator.go#L57-L61

Reply:
I think if we're to keep the generic state of this function, this could be harder to do.